### PR TITLE
IGNITE-14728 Change IGNITE_PDS_WAL_REBALANCE_THRESHOLD from System property to Distributed property

### PR DESCRIPTION
**Pull request by Erixonich**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerPropertiesTest.java [hunk 1]
```diff
@@ -197,4 +199,45 @@ public void testPropertyDefaultQueryTimeout() {
...
+
+        assertDistributedPropertyEquals(WAL_REBALANCE_THRESHOLD_DMS_KEY, newVal, true);
+    }
+
+    /**
+     * Validates that distributed property has specified value across all nodes.
+     * @param propName Distributed property name.
+     * @param expected Expected property value.
+     * @param <T> Property type.
+     */
```
Comment (denis-chudov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerPropertiesTest.java:229]: Please add empty line before paramerters description and a description for `onlyServerMode`.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java [hunk 2]
```diff
@@ -3734,4 +3749,22 @@ public RestoreLogicalState(CheckpointStatus status, WALIterator iterator, long l
...
+            new DistributedConfigurationLifecycleListener() {
+                @Override public void onReadyToRegister(DistributedPropertyDispatcher dispatcher) {
+                    String logMsgFmt = "Historical rebalance WAL threshold [property=%s] changed[oldVal=%s, newVal=%s]";
+                    walRebalanceThreshold.addListener(makeUpdateListener(logMsgFmt, log));
+
+                    dispatcher.registerProperties(walRebalanceThreshold);
+                }
+
+                @Override public void onReadyToWrite() {
+                    setDefaultValue(walRebalanceThreshold, walRebalanceThresholdLegacy, log);
```
[the remark this answers is not in the document]
Comment (Erixonich) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java:3765]: there is an attached to the property on-update listener.
Watch lines above:
```
String logMsgFmt = "Historical rebalance WAL threshold [property=%s] changed[oldVal=%s, newVal=%s]";
walRebalanceThreshold.addListener(makeUpdateListener(logMsgFmt, log));
```

File: modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java [hunk 3]
```diff
@@ -1112,9 +1112,11 @@
 
     /**
      * WAL rebalance threshold.
+     * @deprecated use Distributed MetaStorage property wal.rebalance.threshold.
```
[the remark this answers is not in the document]
Comment (Erixonich) [reply modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java:1115]: ok, I'll do.

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerPropertiesTest.java (hunk 1 above)
Comment (Erixonich) [reply modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerPropertiesTest.java:229]: Will do

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java [hunk 4]
```diff
@@ -359,6 +367,10 @@
     /** Checkpoint frequency deviation. */
     private SimpleDistributedProperty<Integer> cpFreqDeviation;
 
+    /** WAL rebalance threshold. */
+    private final SimpleDistributedProperty<Integer> walRebalanceThreshold =
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java:371]: It's not clear from javadoc what wal rebalance threshold is.

**Pull request closed without merging**